src/chia_log/log_consumer.py

```diff
@@ -83,20 +83,19 @@ def _consume_loop(self):
 class NetworkLogConsumer(LogConsumer):
     """Consume logs over the network"""
 
-    def __init__(self, remote_log_path: Path, remote_user, remote_host):
+    def __init__(self, remote_log_path: PurePath, remote_user, remote_host, remote_platform=OS.LINUX):
```

Comment:
Just noticed, we're missing type annotations for most arguments here, would be great if you can add them.

Let's not have default argument for `remote_platform` so that it's always a required argument. Otherwise it could be wrongly initialized by providing Windows path and not overriding the default platform.

In fact, to avoid that redundancy and potential mismatch you could actually determine the platform directly from the path class now: 

```isinstance(remote_log_path, PureWindowsPath)```

But I'll leave that choice up to you. Maybe it's better for future to have it separate and explicit.

Reply:
I added the type annotations. I would prefer to leave it explicit for now, since PurePosixPath does not distinguish between MacOS and Linux. 

Currently, we don't specifically require the distinction either, but it might be handy in the future.